utils/uri/matcher.go

```diff
@@ -0,0 +1,75 @@
+package uri
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	ErrInvalidURI = errors.New("invalid URI string")
+)
+
+//hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+type URI struct {
```

Comment:
Could you add comments for public methods and structs?

Reply:
Sure 👍